packages/openapi-fetch/test/index.test.ts

```diff
@@ -859,6 +855,138 @@ describe("client", () => {
       });
     });
 
+    describe("content-type", () => {
+      const BODY_ACCEPTING_METHODS = [["PUT"], ["POST"], ["DELETE"], ["OPTIONS"], ["PATCH"]] as const;
+      const ALL_METHODS = [...BODY_ACCEPTING_METHODS, ["GET"], ["HEAD"]] as const;
+
+      const fireRequestAndGetContentType = async (options: {
+        defaultHeaders?: HeadersOptions;
+        method: (typeof ALL_METHODS)[number][number];
+        fetchOptions: FetchOptions<any>;
+      }) => {
+        const client = createClient<any>({ baseUrl, headers: options.defaultHeaders });
+        const { getRequest } = useMockRequestHandler({
+          baseUrl,
+          method: "all",
+          path: "/blogposts-optional",
+          status: 200,
+        });
+        await client[options.method]("/blogposts-optional", options.fetchOptions as any);
+
+        const request = getRequest();
+        return request.headers.get("content-type");
+      };
+
+      it.each(ALL_METHODS)("no content-type for body-less requests - %s", async (method) => {
+        const contentType = await fireRequestAndGetContentType({
+          method,
+          fetchOptions: {},
+        });
+
+        expect(contentType).toBe(null);
+      });
+
+      it.each(ALL_METHODS)("no content-type for `undefined` body requests - %s", async (method) => {
+        const contentType = await fireRequestAndGetContentType({
+          method,
+          fetchOptions: {
+            body: undefined,
+          },
+        });
+
+        expect(contentType).toBe(null);
+      });
+
+      const BODIES = [{ prop: "a" }, {}, "", "str", null, false, 0, 1, new Date("2024-08-07T09:52:00.836Z")] as const;
+      const METHOD_BODY_COMBINATIONS = BODY_ACCEPTING_METHODS.flatMap(([method]) =>
+        BODIES.map((body) => [method, body] as const),
+      );
```

Comment:
@kerwanp Guessing from https://github.com/openapi-ts/openapi-typescript/pull/1825#discussion_r1706738232 you won't like this 🤣 

Happy to ditch the methods, but I'd rather keep the various bodies as this was broken until #1825.

Reply:
Aha I'm not against this when behavior have to be different between cases, the pattern you use looks good to me! The methods constants could be moved at the top of the file to be reused in other tests.

Reply:
Until they are actually reused in other tests, they IMO make more sense when they are collocated with the test itself. It's just less scrolling to understand how they are used. Anyway this is just my opinion I've grown over the years, nothing I would fight for.

I guess we are in agreement that this file already does too much and deserves some larger refactoring and splitting.  Let's maybe open a separate issue and discuss a bit what the desired structure could look like?